src/modules/hip/kernel/resample.hpp

```diff
@@ -0,0 +1,171 @@
+                                             double scale,
+                                             RpptResamplingWindow &window,
+                                             int block)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+
+    int outBlock = id_x * block;
+    if (outBlock >= outEnd)
+        return;
+
+    int blockEnd = std::min(outBlock + block, outEnd);
+    double inBlockRaw = outBlock * scale;
+    int inBlockRounded = static_cast<int>(inBlockRaw);
+    float inPos = inBlockRaw - inBlockRounded;
+    float fscale = scale;
+    float tempBuf[RPPT_MAX_CHANNELS] = {0.0f}; // Considering max channels as 3
+
+    for (int outPos = outBlock; outPos < blockEnd; outPos++, inPos += fscale)
+    {
+        int loc0, loc1;
+        window.input_range(inPos, &loc0, &loc1);
+
+        if (loc0 + inBlockRounded < 0)
+            loc0 = -inBlockRounded;
+        if (loc1 + inBlockRounded > srcDims.x)
+            loc1 = srcDims.x - inBlockRounded;
+        int locInWindow = loc0;
+        float locBegin = locInWindow - inPos;
+        int2 ofs_i2 = make_int2(loc0, loc1);
+        ofs_i2 *= (int2)srcDims.y;
+        int idx = inBlockRounded * srcDims.y;
+
+        for (int inOfs = ofs_i2.x; inOfs < ofs_i2.y; inOfs += srcDims.y, locBegin++)
+        {
+            float w = window(locBegin);
+            for (int c = 0; c < srcDims.y; c++)
+                tempBuf[c] += srcPtr[idx + inOfs + c] * w;
+        }
+        int dstLoc = outPos * srcDims.y;
+        for (int c = 0; c < srcDims.y; c++)
+            dstPtr[dstLoc + c] = tempBuf[c];
+    }
+}
+
+// -------------------- Set 1 - resample kernels executor --------------------
+
+RppStatus hip_exec_resample_tensor(Rpp32f *srcPtr,
+                                   RpptDescPtr srcDescPtr,
+                                   Rpp32f *dstPtr,
+                                   RpptDescPtr dstDescPtr,
+                                   Rpp32f *inRateTensor,
+                                   Rpp32f *outRateTensor,
+                                   Rpp32s *srcDimsTensor,
+                                   RpptResamplingWindow &window,
+                                   rpp::Handle& handle)
+{
+    int batchSize = handle.GetBatchSize();
+
+    for(int i = 0; i < batchSize; i++)
+    {
+        float inRate = inRateTensor[i];
+        float outRate = outRateTensor[i];
+        int srcLength = srcDimsTensor[i * 2];
+        int numChannels = srcDimsTensor[i * 2 + 1];
+        if (inRate == outRate) // No need of Resampling, do a direct memcpy
+            hipMemcpy(dstPtr, srcPtr, srcLength * numChannels * sizeof(float), hipMemcpyDeviceToDevice);
+        else
+        {
+            int outEnd = std::ceil(srcLength * outRate / inRate);
+            double scale = static_cast<double>(inRate) / outRate;
+            int block = 256; // 1 << 8
+            int length = (outEnd / block) + 1;
+            int globalThreads_x = length;
```

Comment:
Pls remove extra variable definitions. You can use LOCAL_THREADS_X_1DIM for block size below, and directly assign int globalThreads_x = (outEnd / LOCAL_THREADS_X_1DIM) + 1;